mango-core/src/main/java/org/calrissian/mango/domain/BaseAttributeStore.java

```diff
@@ -75,8 +74,19 @@ public void putAll(Iterable<Attribute> attributes) {
     }
 
     @Override
+    public <T> Attribute<T> get(String key, Class<T> clazz) {
+        Collection<Attribute> attrs = getAll(key);
+        for(Attribute attr: attrs) {
+            if(attr.getValue().getClass().isAssignableFrom(clazz))
+                return attr;
+        }
+
+        return null;
+    }
+
+    @Override
     public Set<String> keys() {
-        return attributes.keySet();
+        return new HashSet(attributes.keySet());
```

Comment:
No because I need to modify it. I don't like the use of unmodifiable
everywhere because it's gotten me into nasty unsupported operations and the
performance hinderance of having to copy twice is just too much.

On Wed, Mar 11, 2015 at 2:34 AM, eawagner <notifications@github.com> wrote:

> In
> mango-core/src/main/java/org/calrissian/mango/domain/BaseAttributeStore.java
> <https://github.com/calrissian/mango/pull/178#discussion_r26191468>:
>
> >      public Set<String> keys() {
> > -        return attributes.keySet();
> > +        return new HashSet(attributes.keySet());
>
> Just use Collections.unmodifiableSet() instead of copying the set.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/calrissian/mango/pull/178/files#r26191468>.
>